src/ConfigAggregator.php

```diff
@@ -285,14 +292,31 @@ private function cacheConfig(array $config, null|string $cachedConfigFile): void
         $this->writeCache($cachedConfigFile, $contents, $mode);
     }
 
+    /**
+     * @param list<PreProcessorCallable|class-string> $processors
+     * @param ProviderIterable $providers
+     * @return ProviderIterable
+     */
+    private function preProcessProviders(array $processors, iterable $providers): iterable
+    {
+        foreach ($processors as $processor) {
+            /** @var PreProcessorCallable $processorCallable */
```

Comment:
Can we somehow move this to the method declaration?

Reply:
iirc, I tried this, but psalm doesn't seem to like a conditional return using custom types